### ISSUE TITLE
can't redefine non-configuratable property "__id" error in a use of this in a constructor

### ISSUE THREAD

**Issue by masuhar** [prg-titech/Kanon#11]
An assignment of this in a constructor to another object will cause an error `can't redefine non-configuratable property "__id" `
The following snippet is an example.
```
class Node {
    constructor(prev) {
        prev.a = this;
    }
}

new Node({a:null});
```
![image](https://user-images.githubusercontent.com/7893790/29911892-a6f0e6f4-8e6a-11e7-9587-dfb531f48a0f.png)

**Comment by akiou** [comment]
Thank you for your comment.
I have fixed this issue. So please ask your confirmation.

**Comment by masuhar** [comment]
Thanks.  It was fixed. 

(因みにissueに書かれた問題を修正した場合は、commit messageの方に「fixes #11」のように書いておくと、commit記録がissueのコメントに書かれ、issueがcloseされます。issueはreopenできるので、修正できたと思った開発者は自分でcloseするのが一般的な流儀のようです。)
